### ISSUE TITLE
DialogUXStateAggregator goes back to Listening state

### ISSUE THREAD

**Issue by jbfitb** [alexa/avs-device-sdk#1783]
## Briefly summarize your issue:
I noticed that our app would go from `Thinking` to `Listening` and back to `Idle`. I was very confused until I saw this line:
https://github.com/alexa/avs-device-sdk/blob/master/AVSCommon/AVS/src/DialogUXStateAggregator.cpp#L121

I believe
```
            case AudioInputProcessorObserverInterface::State::BUSY:
                setState(DialogUXStateObserverInterface::DialogUXState::LISTENING);
```

should be

```
            case AudioInputProcessorObserverInterface::State::BUSY:
                setState(DialogUXStateObserverInterface::DialogUXState::THINKING);
```

Logs:

```
DialogUXStateAggregator:onRequestProcessingStartedLambda:currentState=LISTENING
DialogUXStateAggregator:setState:from=LISTENING,to=THINKING
AudioInputProcessor:executeOnDialogUXStateChanged:newState=THINKING
AudioInputProcessor:stopCapture::stopCaptureInvoked
AudioInputProcessor:executeStopCapture::creatingStopCaptureLambda
AudioInputProcessor:stopCapture:stopImmediately=false
AudioInputProcessor:setState:from=RECOGNIZING,to=BUSY
DialogUXStateAggregator:setState:from=THINKING,to=LISTENING
```

**Comment by jbfitb** [comment]
Right, so what I think is happening is this:

1. Start a new utterance: `Recognize` is called on `AudioInputProcessor` [which sets the state to `RECOGNIZE`](https://github.com/alexa/avs-device-sdk/blob/master/CapabilityAgents/AIP/src/AudioInputProcessor.cpp#L968)
1. `DialogUXStateAggregator::onStateChange` is invoked with `RECOGNIZE`. [This sets the dialog state to `LISTENING`](https://github.com/alexa/avs-device-sdk/blob/master/AVSCommon/AVS/src/DialogUXStateAggregator.cpp#L114)
1. Audio upload happens... time passes...
1. `RequestProcessingStarted` directive is received from the cloud. [This sets the state in `DialogUXStateAggregator` to `THINKING`](https://github.com/alexa/avs-device-sdk/blob/master/AVSCommon/AVS/src/DialogUXStateAggregator.cpp#L203)
1. Then we get a `StopCapture` directive. [This sets the AudioInputProcessor state to `BUSY`](https://github.com/alexa/avs-device-sdk/blob/master/CapabilityAgents/AIP/src/AudioInputProcessor.cpp#L1162) which in turn, [sends it to the `DialogUXStateAggregator`](https://github.com/alexa/avs-device-sdk/blob/master/CapabilityAgents/AIP/src/AudioInputProcessor.cpp#L1315).  [That invokes `onStateChange`](https://github.com/alexa/avs-device-sdk/blob/master/AVSCommon/AVS/src/DialogUXStateAggregator.cpp#L104) of `DialogUXStateAggregator` again which [sets the state back to `LISTENING`](https://github.com/alexa/avs-device-sdk/blob/master/AVSCommon/AVS/src/DialogUXStateAggregator.cpp#L121)

[I believe this line of code is incorrect](https://github.com/alexa/avs-device-sdk/blob/master/AVSCommon/AVS/src/DialogUXStateAggregator.cpp#L121)

This

```
            case AudioInputProcessorObserverInterface::State::BUSY:
                setState(DialogUXStateObserverInterface::DialogUXState::LISTENING);
```

should be
```
            case AudioInputProcessorObserverInterface::State::BUSY:
                setState(DialogUXStateObserverInterface::DialogUXState::**THINKING**);
```

since "BUSY" is the cloud state of processing and `LISTENING` is while we're uploading data.

**Comment by jbfitb** [comment]
One thing to add: both `AudioInputProcessor` and `DialogUXStateAggregator` use executor services so there could be race condition between the `StopCapture` and `RequestProcessingStarted` directives. They may not always be processed in the same order.

**Comment by kjkh** [comment]
Thanks for the information @jbfitb. It's interesting that that `RequestProcessingStarted` is received **before** `StopCapture`, since that is not expected. We expect the reverse. However, even if the order of the directives are as we expect, there can be possible race conditions between the executors, as you suggested. This suggests that DUXSA needs to be more robust. We are looking into a fix for this.

**Comment by kclchan** [comment]
This has been fixed in v1.21.  Thanks!
